### DISCUSSION TITLE
Accessing full binary path?

### DISCUSSION THREAD

**Discussion by danpf** [williamboman/nvim-lsp-installer#770] [Q&A] [answered]
I try to keep all of the things I install for vim separate from my development environment.  Using python as an example, I do this by installing npm and python black in a conda `nvimenv`, and then I develop in a separate conda environment.

Unfortunately, in the case of `pyright-langserver`, it requires `node` to be installed in your working path in vim.  To get around this I have the following setup.  basically it's: use conda run to run a command to get an environment with node in it, then use the full path to call upon the pyright-server.  Overall, it's fine and works, but I feel like this should be dooable without so much hoopla.  I have seen this come up before in some issues, but I wanted to just bring this up as an example, and to ask if there's a better way.  (i'm only showing pyright here but this is a problem with other servers that have outside dependencies as well)

it also gets a little more murky because lsp-installer controls the install, but lspconfig controls the binary name.  I guess this would be easier if this repo would maintain a table of all binaries that have been installed for a server, and where they live in a more accessible fashion. maybe even a map of base-executable: full-path-executable would be nice. Just thinking out loud with that though.

```
local find_in_path = function(basename, path)
  local split_path = vim.split(path, ":")
  local found_exe = nil
  for _, p in ipairs(split_path) do
    local working_fn = p .. '/' .. basename
    local fn_found = io.open(p .. '/' .. basename, "r")
    if fn_found ~= nil then
      io.close(fn_found)
      found_exe = working_fn
    end
  end
  return found_exe
end

...

  local lsp_installer = require "nvim-lsp-installer"
  local newattach = function(client, bufnr)
    if client.name == "pyright" then
      local runprecmd = {"conda", "run", "-n", "nvimenv"}
      local server_exe = client.config.cmd[1]
      for _, v in ipairs(client.config.cmd) do
        if v == server_exe then
          table.insert(runprecmd, find_in_path(v, client.config.cmd_env.PATH))
        else
          table.insert(runprecmd, v)
        end
      end
      client.config.cmd = runprecmd
      attach(client, bufnr)
    end
  end

```

as an aside, the simplest answer is to just change the cmd_env when attaching, instead of everything that i've written above...

but when you do this, you actually include the development environment in the command, which is something I'd like to avoid.

for reference this would just look like this

```
 ...
      client.config.cmd_env.PATH = "/home/me/miniconda3/envs/nvimenv/bin:" .. vim.env.PATH
      attach(client, bufnr)
....
```

cheers and thanks for your work on this repo, it's very useful!

**Comment by williamboman** [comment, marked as answer]
Hello! There's no nice way of doing this currently. I've been working on a refactoring that will symlink binaries to a single, common, location (and prepend that location to the `PATH` environment of the entire neovim instance). I believe that would more or less solve your problem! Also see 

- https://github.com/Pocco81/dap-buddy.nvim/issues/71#issuecomment-1106527425
- https://github.com/Pocco81/dap-buddy.nvim/issues/71#issuecomment-1167689128

**Reply by williamboman** [reply]
Cross-posting this here: https://github.com/williamboman/nvim-lsp-installer/issues/95#issuecomment-1176925455
